src/app/index.ts

```diff
@@ -156,41 +219,84 @@ module.exports = class TheiaPlugin extends Base {
             pluginSourcePath: pluginName + '-' + pluginType + '-plugin.ts',
             pluginDistPath: pluginName + '-' + pluginType + '-plugin.js',
             isFrontend: (options.pluginType === 'frontend'),
-            isBackend: (options.pluginType === 'backend')
+            isBackend: (options.pluginType === 'backend'),
+            template: options.template,
+            sample: options.sample,
         };
         options.params = this.params;
 
     }
 
     writing() {
+        this._writeBase();
+        switch (this.params.template) {
+            case 'hello-world':
+                this._writeMain('hello-world.ts');
+                break;
+            case 'skeleton':
+                this._writeMain('empty/index.ts');
+                break;
+            case 'samples':
+                this._writeSample(this.params.sample!);
+                break;
+        }
+
+    }
+
+    private _writeSample(sample: string): void {
+        if (!sample) {
+            sample = (<any>this.options).sample;
+        }
+        const path = 'samples/' + sample + '/';
+
+        this._writeMain(path + 'index.ts');
+        try {
+            this.fs.copyTpl(
+                this.templatePath(path) + '!(index.ts)',
+                this.destinationPath('src'),
+                { params: this.params },
+                {},
+                { globOptions: { dot: true } }
+            );
+        } catch (e) {
+            // ignore copy errors if template doesn't have others files
```

Comment:
we should log it to debug level to avoid silent catch or avoid exception by checking file exists